.github/workflows/build.yml

```diff
@@ -0,0 +1,56 @@
+name: build
+on:
+  workflow_dispatch:
+  push:
+    branches: [master, "release-**"]
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest]
+        ghc: ["8.6.5", "8.8.4", "8.10.2"]
+    name: build - ${{ matrix.ghc }} - ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Install LLVM (for running tests)
+        run: sudo apt-get -y install llvm
+
+      - uses: actions/setup-haskell@v1
+        id: setup-haskell
+        with:
+          ghc-version: ${{ matrix.ghc }}
+
+      - uses: actions/cache@v2
+        name: Cache cabal store
+        with:
+          path: |
+            ${{ steps.setup-haskell.outputs.cabal-store }}
+            dist-newstyle
+          key: cabal-${{ runner.os }}-${{ matrix.ghc }}
```

Comment:
You really want `cabal-${{ runner.os}}-${{ matrix.ghc }}-${{ github.sha }}` here with the fallback key being `cabal-${{ runner.os}}-${{ matrix.ghc }}-` (note the ending `-`). Otherwise the cache will never update and build times will start to get longer as time goes on. Especially if dependencies are ever changed.

Reply:
Oh, yikes. Will do!